AsyncDisplayKit/Layout/ASLayoutSpec.mm

```diff
@@ -128,6 +146,27 @@ - (NSArray *)children
   return self.layoutChildren[kDefaultChildrenKey];
 }
 
+
+#pragma mark - ASEnvironment
+
+- (ASEnvironmentState)environmentState
+{
+  return _environmentState;
+}
+
+- (void)setEnvironmentState:(ASEnvironmentState)environmentState
+{
+  _environmentState = environmentState;
+}
+
+- (BOOL)supportsUpwardPropagation
+{
+  return YES;
```

Comment:
Add a comment that says that two subclasses override this method to return NO, because upward propagation is not enabled if a layout specification has more than one child, and these are currently the specifications that are known to have more than one.